### PR TITLE
CSP-649 AAN HUB API POST Employer To Take Profile Array

### PR DESCRIPTION
**Pull request by YuvrajPatil-DFE**

### CODE REVIEW

File: src/SFA.DAS.AANHub.Application/Employers/Commands/CreateEmployerMember/CreateEmployerMemberCommandValidator.cs [hunk 1]
```diff
@@ -1,31 +1,46 @@
...
+
+        RuleFor(c => c.ProfileValues)
+            .NotEmpty()
+            .WithMessage(ProfileValuesMustNotBeEmptyErrorMessage)
+            .ForEach(x => x.SetValidator(new ProfileValueValidator()))
+            .MustAsync(async (profileValues, _) =>
+            {
+                var profiles = await _profilesReadRepository.GetProfilesByUserType(MembershipUserType.Employer);
+                var profileIds = profiles.Select(profile => profile.Id);
+                var b = profileValues.Select(v => v.Id).All(i => profileIds.Contains(i));
```
Comment (DevenKShah) [inline src/SFA.DAS.AANHub.Application/Employers/Commands/CreateEmployerMember/CreateEmployerMemberCommandValidator.cs:40]: Simplified `profileValues.All(p => profileIds.Contains(p.Id))`

File: src/SFA.DAS.AANHub.Application/Employers/Commands/CreateEmployerMember/CreateEmployerMemberCommandValidator.cs [hunk 2]
```diff
@@ -1,31 +1,46 @@
...
-            .MaximumLength(250);
-
         RuleFor(c => c.AccountId)
             .NotEmpty();
+
+        RuleFor(c => c.ProfileValues)
+            .NotEmpty()
+            .WithMessage(ProfileValuesMustNotBeEmptyErrorMessage)
+            .ForEach(x => x.SetValidator(new ProfileValueValidator()))
+            .MustAsync(async (profileValues, _) =>
```
Comment (DevenKShah) [inline src/SFA.DAS.AANHub.Application/Employers/Commands/CreateEmployerMember/CreateEmployerMemberCommandValidator.cs:36]: There is no coverage for this validation, empty and invalid ids both needs to be covered with positive tests.

File: src/SFA.DAS.AANHub.Application/Employers/Commands/CreateEmployerMember/CreateEmployerMemberCommand.cs [hunk 3]
```diff
@@ -28,6 +27,9 @@ public class CreateEmployerMemberCommand : CreateMemberCommandBase, IRequest<Val
             MemberId = command.MemberId,
             AccountId = command.AccountId,
             UserRef = command.UserRef
-        }
+        },
+        MemberProfiles = command.ProfileValues.Select(p => ProfileConverter(p, command.MemberId)).ToList()
     };
+
+    private static MemberProfile ProfileConverter(ProfileValue source, Guid memberId) => new() { MemberId = memberId, ProfileId = source.Id, ProfileValue = source.Value };
```
Comment (DevenKShah) [inline src/SFA.DAS.AANHub.Application/Employers/Commands/CreateEmployerMember/CreateEmployerMemberCommand.cs:34]: Coverage missing

File: src/SFA.DAS.AANHub.Application/Employers/Commands/CreateEmployerMember/CreateEmployerMemberCommandValidator.cs (hunk 2 above)
Comment (YuvrajPatil-DFE) [reply src/SFA.DAS.AANHub.Application/Employers/Commands/CreateEmployerMember/CreateEmployerMemberCommandValidator.cs:36]: I think this one is covered as I can see the errors populating in testResult

**Pull request merged**